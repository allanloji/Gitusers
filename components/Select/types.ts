type Option = {
  label: string;
  value: string;
};

type SelectProps = {
  name: string;
  options: Option[];
  onChange: (value: string) => void;
};

export type { SelectProps };
