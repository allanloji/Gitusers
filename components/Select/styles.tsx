import styled from '@emotion/styled';

const Label = styled.label`
  color: white;
  display: flex;
  flex-direction: column;
  font-size: 20px;
`;

const StyledSelect = styled.select`
  margin-top: 1rem;
  height: 50px;
  color: white;
  border: none;
  background-color: rgb(18, 18, 18);
  outline: none;
  padding: 0.5rem;
  font-size: 20px;
`;
export { Label, StyledSelect };
