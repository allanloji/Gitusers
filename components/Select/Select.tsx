import React from 'react';

import { Label, StyledSelect } from './styles';
import { SelectProps } from './types';

/**
 * Select
 */
function Select({ label, options, name, onChange }: SelectProps) {
  return (
    <React.Fragment>
      <StyledSelect name={name} id={name} onChange={e => onChange(e.target.value)}>
        {options.map(option => (
          <option key={option.label} value={option.value}>
            {option.label}
          </option>
        ))}
      </StyledSelect>
    </React.Fragment>
  );
}

export default Select;
