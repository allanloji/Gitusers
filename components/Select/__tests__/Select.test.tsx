import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Select from '..';

describe('Select', () => {
  test('Should render', () => {
    const onChange = jest.fn();
    const options = [
      { label: 'Any order', value: '' },
      { label: 'Asc', value: 'asc' },
      { label: 'Desc', value: 'desc' },
    ];

    const { container } = render(<Select name='perPage' onChange={onChange} options={options} />);

    expect(container).toMatchSnapshot();
  });

  test('Should execute onChange function', () => {
    const onChange = jest.fn();
    const options = [
      { label: 'Any order', value: '' },
      { label: 'Asc', value: 'asc' },
      { label: 'Desc', value: 'desc' },
    ];

    const { getByRole } = render(<Select name='perPage' onChange={onChange} options={options} />);

    const input = getByRole('combobox');
    userEvent.selectOptions(input, 'Asc');

    expect(onChange).toHaveBeenCalledWith('asc');
  });
});
