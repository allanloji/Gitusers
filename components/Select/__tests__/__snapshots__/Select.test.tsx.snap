// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should render 1`] = `
<div>
  <select
    class="css-1fhsdaz"
    id="perPage"
    name="perPage"
  >
    <option
      value=""
    >
      Any order
    </option>
    <option
      value="asc"
    >
      Asc
    </option>
    <option
      value="desc"
    >
      Desc
    </option>
  </select>
</div>
`;
